### PR TITLE
mypy fixes

### PR DESCRIPTION
**Pull request by synarete**
Fix mypy errors in order to enable CI via github's actions. 
For now, do the minimal required changes to python code base in order to have minimal CI ob (check only).

### CODE REVIEW

File: testcases/consistency/test_consistency.py [hunk 1]
```diff
@@ -10,11 +10,12 @@
 import os
 import sys
 import pytest
+import typing
 
 test_string = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
 
 # Use a global test_info to get a better output when running pytest
-test_info = {}
+test_info: dict[typing.Any, typing.Any] = {}
```
[the remark this answers is not in the document]
Comment (synarete) [reply testcases/consistency/test_consistency.py:18]: `mypy` require explicit type annotation for this global variable. It should probably be `dict[str, typing.Any]` but I decided to move the this precise type annotation to a following PR.

[the remark this answers is not in the document]
Comment (phlogistonjohn) [reply testcases/consistency/test_consistency.py:18]: Interesting, thanks.

**Comment by phlogistonjohn** [comment]
Looks like the host is on an older version of python:
```
==================================== ERRORS ====================================
__________ ERROR collecting testcases/consistency/test_consistency.py __________
testcases/consistency/test_consistency.py:18: in <module>
    test_info: dict[typing.Any, typing.Any] = {}
E   TypeError: 'type' object is not subscriptable
___________ ERROR collecting testcases/smbtorture/test_smbtorture.py ___________
testcases/smbtorture/test_smbtorture.py:22: in <module>
    test_info: dict[typing.Any, typing.Any] = {}
E   TypeError: 'type' object is not subscriptable
```

You'll probably need `typing.Dict[typing.Any, typing.Any] `.  You may want to set up your tox envs so that you also test with an older version of python to avoid these kinds of things. Create a environment like `mypy-py39` or whatever. Let me know if you need additional help with that.

**Comment by synarete** [comment]
> Looks like the host is on an older version of python:
> 
> ```
> ==================================== ERRORS ====================================
> __________ ERROR collecting testcases/consistency/test_consistency.py __________
> testcases/consistency/test_consistency.py:18: in <module>
>     test_info: dict[typing.Any, typing.Any] = {}
> E   TypeError: 'type' object is not subscriptable
> ___________ ERROR collecting testcases/smbtorture/test_smbtorture.py ___________
> testcases/smbtorture/test_smbtorture.py:22: in <module>
>     test_info: dict[typing.Any, typing.Any] = {}
> E   TypeError: 'type' object is not subscriptable
> ```
> 
> You'll probably need `typing.Dict[typing.Any, typing.Any] `. You may want to set up your tox envs so that you also test with an older version of python to avoid these kinds of things. Create a environment like `mypy-py39` or whatever. Let me know if you need additional help with that.

I developed and checked this PR on CentOS-stream9 and everything was OK. But now I realize that the problem may be deeper: `tox` needs [Python-3.7 or higher](https://tox.wiki/en/latest/installation.html) where centos-ci still uses CentOS8 with `Python 3.6.8`.

@anoopcs9 Are there any plans to move centos-ci to `CentOS-stream9` any time soon?

**Comment by anoopcs9** [comment]
> > Looks like the host is on an older version of python:
> > ```
> > ==================================== ERRORS ====================================
> > __________ ERROR collecting testcases/consistency/test_consistency.py __________
> > testcases/consistency/test_consistency.py:18: in <module>
> >     test_info: dict[typing.Any, typing.Any] = {}
> > E   TypeError: 'type' object is not subscriptable
> > ___________ ERROR collecting testcases/smbtorture/test_smbtorture.py ___________
> > testcases/smbtorture/test_smbtorture.py:22: in <module>
> >     test_info: dict[typing.Any, typing.Any] = {}
> > E   TypeError: 'type' object is not subscriptable
> > ```
> >  
> > You'll probably need `typing.Dict[typing.Any, typing.Any] `. You may want to set up your tox envs so that you also test with an older version of python to avoid these kinds of things. Create a environment like `mypy-py39` or whatever. Let me know if you need additional help with that.
> 
> I developed and checked this PR on CentOS-stream9 and everything was OK. But now I realize that the problem may be deeper: `tox` needs [Python-3.7 or higher](https://tox.wiki/en/latest/installation.html) where centos-ci still uses CentOS8 with `Python 3.6.8`.
> 
> @anoopcs9 Are there any plans to move centos-ci to `CentOS-stream9` any time soon?

Python 3.8, 3.9 and 3.11 are available with CentOS Stream 8 but are not installed yet on client machines. We could install the required python version and teach tox and friends to execute based on it. Since these are executed on client VMs any changes would go into [sit-environment](https://github.com/samba-in-kubernetes/sit-environment) repository where necessary [ansible roles](https://github.com/samba-in-kubernetes/sit-environment/tree/main/vagrant/ansible/roles/client.prep) are kept for automation.

@phlogistonjohn @spuiuk What do you think?

**Comment by spuiuk** [comment]
> > I developed and checked this PR on CentOS-stream9 and everything was OK. But now I realize that the problem may be deeper: `tox` needs [Python-3.7 or higher](https://tox.wiki/en/latest/installation.html) where centos-ci still uses CentOS8 with `Python 3.6.8`.
> > @anoopcs9 Are there any plans to move centos-ci to `CentOS-stream9` any time soon?
> 
> Python 3.8, 3.9 and 3.11 are available with CentOS Stream 8 but are not installed yet on client machines. We could install the required python version and teach tox and friends to execute based on it. Since these are executed on client VMs any changes would go into [sit-environment](https://github.com/samba-in-kubernetes/sit-environment) repository where necessary [ansible roles](https://github.com/samba-in-kubernetes/sit-environment/tree/main/vagrant/ansible/roles/client.prep) are kept for automation.
> 
> @phlogistonjohn @spuiuk What do you think?

These packages are only required for tox which is only used for checking the code in the CI pipeline or on our dev environments which is primarily F38 and CentOS 9. Will this tool be ever be run on the client? I don't really have any objection installing it on the client but I wonder if that is the right environment on which to run it.

**Comment by synarete** [comment]
> > > I developed and checked this PR on CentOS-stream9 and everything was OK. But now I realize that the problem may be deeper: `tox` needs [Python-3.7 or higher](https://tox.wiki/en/latest/installation.html) where centos-ci still uses CentOS8 with `Python 3.6.8`.
> > > @anoopcs9 Are there any plans to move centos-ci to `CentOS-stream9` any time soon?
> > 
> > 
> > Python 3.8, 3.9 and 3.11 are available with CentOS Stream 8 but are not installed yet on client machines. We could install the required python version and teach tox and friends to execute based on it. Since these are executed on client VMs any changes would go into [sit-environment](https://github.com/samba-in-kubernetes/sit-environment) repository where necessary [ansible roles](https://github.com/samba-in-kubernetes/sit-environment/tree/main/vagrant/ansible/roles/client.prep) are kept for automation.
> > @phlogistonjohn @spuiuk What do you think?
> 
> These packages are only required for tox which is only used for checking the code in the CI pipeline or on our dev environments which is primarily F38 and CentOS 9. Will this tool be ever be run on the client? I don't really have any objection installing it on the client but I wonder if that is the right environment on which to run it.

Nope -- tox is needed only as part of dev env (to check the code); never in production. IMHO, we may have those checks only as part of github's CI actions workflow.

**Comment by phlogistonjohn** [comment]
I suggest installing python-3.Y on whatever node(s) run tox. The Y should be whatever the oldest python version the real code needs to work on.  The node might be github runner or your local machine. Once you have a python interpreter of the correct version tox should take care of the rest of the dependencies.  Do note that I am not sure what happens if you run *only* mypy - I'm not sure if the original problem would be  surfaced by that.... something to try out.

**Comment by phlogistonjohn** [comment]
What I forgot to write is: you need a matching tox environment like py39, py36, mypy-py38, or similar

**Comment by spuiuk** [comment]
@synarete make test returns a failure which seems to be related to these changes. 

```

TASK [client.test : Run tests] *************************************************
fatal: [client0]: FAILED! => {
    "changed": true,
    "cmd": [
        "make",
        "test"
    ],
    "delta": "0:00:00.356137",
    "end": "2023-07-16 14:26:29.193209",
    "rc": 2,
    "start": "2023-07-16 14:26:28.837072"
}

STDOUT:

============================= test session starts ==============================
platform linux -- Python 3.6.8, pytest-3.4.2, py-1.5.3, pluggy-0.6.0 -- /usr/libexec/platform-python
cachedir: .pytest_cache
rootdir: /root/samba-integration-tests, inifile:
collecting ... collected 1 item / 2 errors

==================================== ERRORS ====================================
__________ ERROR collecting testcases/consistency/test_consistency.py __________
testcases/consistency/test_consistency.py:18: in <module>
    test_info: dict[typing.Any, typing.Any] = {}
E   TypeError: 'type' object is not subscriptable
___________ ERROR collecting testcases/smbtorture/test_smbtorture.py ___________
testcases/smbtorture/test_smbtorture.py:22: in <module>
    test_info: dict[typing.Any, typing.Any] = {}
E   TypeError: 'type' object is not subscriptable
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 2 error in 0.10 seconds ============================

```

**Comment by synarete** [comment]
> @synarete make test returns a failure which seems to be related to these changes.
> 
> ```
> 
> TASK [client.test : Run tests] *************************************************
> fatal: [client0]: FAILED! => {
>     "changed": true,
>     "cmd": [
>         "make",
>         "test"
>     ],
>     "delta": "0:00:00.356137",
>     "end": "2023-07-16 14:26:29.193209",
>     "rc": 2,
>     "start": "2023-07-16 14:26:28.837072"
> }
> 
> STDOUT:
> 
> ============================= test session starts ==============================
> platform linux -- Python 3.6.8, pytest-3.4.2, py-1.5.3, pluggy-0.6.0 -- /usr/libexec/platform-python
> cachedir: .pytest_cache
> rootdir: /root/samba-integration-tests, inifile:
> collecting ... collected 1 item / 2 errors
> 
> ==================================== ERRORS ====================================
> __________ ERROR collecting testcases/consistency/test_consistency.py __________
> testcases/consistency/test_consistency.py:18: in <module>
>     test_info: dict[typing.Any, typing.Any] = {}
> E   TypeError: 'type' object is not subscriptable
> ___________ ERROR collecting testcases/smbtorture/test_smbtorture.py ___________
> testcases/smbtorture/test_smbtorture.py:22: in <module>
>     test_info: dict[typing.Any, typing.Any] = {}
> E   TypeError: 'type' object is not subscriptable
> !!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
> =========================== 2 error in 0.10 seconds ============================
> ```

@spuiuk As discussed above, the root cause of those failures is related to using Python 3.6.8 on CI machine. I am trying to figure out if there is an option to fix it via `tox` setting -- will update once I have concrete findings.

**Comment by spuiuk** [comment]
ahh, sorry - my memory got reset over the weekend.

**Comment by spuiuk** [comment]
The latest failure is at
error: samba3.smb2.create.multi [
This is a known issue  in the smbtorture tests in the sit-test-cases and not related to the changes here. 

The CI tests can be assumed to have passed.

**Review by anoopcs9 — APPROVED** [review]
lgtm.

**Review by spuiuk — APPROVED** [review]
CentosCI has some ongoing issues which caused failure in the CI run for this set of patches.
However based on the last run of sit-testing and test results from the actions tab in Shachar's test repo - acking it.
https://github.com/synarete/sit-test-cases/actions/runs/5585488855/jobs/10208292010

**Pull request merged**